src/routes/api/api-router.ts

```diff
@@ -82,16 +82,23 @@ ApiRouter.post(
 		const offset = Number(req.body.offset) || 0;
 		// only resync installations whose "updatedAt" date is older than x seconds
 		const inactiveForSeconds = Number(req.body.inactiveForSeconds) || undefined;
+		// How far back to fetch commit history(main and branch) in milliseconds.
+		const commitTimeLimit = Number(req.body.commitTimeLimit) || undefined;
 
 		if (!statusTypes && !installationIds && !limit && !inactiveForSeconds){
 			res.status(400).send("please provide at least one of the filter parameters!");
 			return;
 		}
 
+		if (commitTimeLimit !== undefined && (commitTimeLimit <= 0 || isNaN(commitTimeLimit))){
+			res.status(400).send("Invalid commitTimeLimit value, please enter a natural number.");
+			return;
+		}
+
 		const subscriptions = await Subscription.getAllFiltered(installationIds, statusTypes, offset, limit, inactiveForSeconds);
 
 		await Promise.all(subscriptions.map((subscription) =>
-			findOrStartSync(subscription, req.log, syncType)
+			findOrStartSync(subscription, req.log, syncType, commitTimeLimit)
```

Comment:
This I'm not sure I agree with, now that I think about it.  In the UI, it'll be something like "backfill from 7 days ago", which is fine, but I think that should be a date coming in specifically as it should be based on the user's current timezone and not the server's.

That means that the 'timeLimit' in this case is specifically a date which should be filtered down to the graphql query which is a GitTimestamp (a version of date).

Reply:
Have converted to Dates when appropriate 🤟